csrc/index_compute.cpp

```diff
@@ -3291,81 +3291,593 @@ Val* Index::eye(
+// the user. For example, it is possible that there is a 1D tensor of shape
+// 1000000 and the user wants to view it as (1000, 1000) and load a tile of (8,
+// 8) from it. Another example is, the user has a tensor of shape (1000, 1000),
+// and as long as the tensor is contiguous, the user can view it as a flattened
+// shape 1000000 tensor, and load a tile of 8 from it. The dimensionality of the
+// tensor in the eyes of TMA is determined by the schedule of the tensor.
+// However, the dimensionality of the tensor in the eyes of TMA can not be
+// smaller than the physical dimensionality of the tensor. For example, if the
+// user has a tensor of shape (128, 128, 128, 128, 128, 128), and the tensor is
+// fully discontiguous, then it is not possible to view it as a <= 5D tensor,
+// thus TMA can not be used to load/store a tile from it.
+//
+// The gmem tensor of TMA store, or the gmem tensor of TMA load when replayed as
+// consumer, must be scheduled like this:
+//
+//                         ---------------------
+//                         | Allocation Domain |
+//                         ---------------------
+//                           | | | | | | | | |
+//                         IterDomain Expressions
+//                             | | | | | | |
+//                         ---------------------
+//                         |  I1           I2  |
+//                         ---------------------
+//                            |            |
+//                          split        split
+//                          /   \        /   |
+//                         I3   I4      I5   I6
+//                         |    |       |    |
+//                        ...   |      ... split
+//                              |          /   |
+//                              |         I7   I8
+//                              |         |    |
+//                              |         |   ...
+//                              |         |
+//                         IterDomain Expressions
+//                          |    |    |    |    |
+//                         Bulk Bulk Bulk Bulk Bulk
+//
+// To schedule TMA, the overall process is:
+//
+// First, we need to specify how TMA should view the tensor. In the above
+// diagram, TMA view the tensor as a 2D tensor [I1, I2]. Note that the
+// IterDomain expressions between the allocation domain and [I1, I2] must be
+// compatible with the allocation domain, for example, we can not merge
+// discontiguous IterDomains, and we can not have indivisible splits either.
+//
+// Second, we need to tile each dimension of TMA's view of the tensor. If we
+// want a contiguous box dim (to be described below), then we do one split to
+// get the box. If we want a strided box dim, then we do two splits, the first
+// split is to get the box, and the second split is to get the stride. In the
+// above diagram, I1 has a contiguous box dim, and I2 has a strided box dim.
```

Comment:
Where's the definition of the box dim and the strided box dim?

Reply:
It is in Definition 3: "boxId"

Reply:
Updated the comment to refer to Definition 3